evaluator/src/main/kotlin/Rule.kt

```diff
@@ -106,18 +106,18 @@ abstract class Rule(
     }
 
     /**
-     * A [RuleMatcher] that checks whether a [label] exists in the [ORT result][OrtResult.labels]. If [value] is null
-     * the value of the label is ignored.
+     * A [RuleMatcher] that checks whether a label denoted by the given [key] exists and optionally (in case the given
+     * [value] is not null) if [value] is amongst the label values of that label.
```

Comment:
I believe that description is misleading now. It sounds like if the label's value is e.g. a comma-separated string, then that string is interpreted as a list, and it is checked whether `value` is part of that list. But that's not how it's implemented.

Reply:
> I wonder whether there is actually any use case for `hasLabel()`.

I believe there is. Lazy people might simply want to check for the presence of a `DO_SOMETHING` label instead of requiring the value to be "true" / "false" and parsing it as a boolean.

Reply:
Let me try to clarify what I meant. 

Can we replace labelContains() and hasLabel() (see Rule.kt) with the following one in OrtResult plus 
an identical one (which returns a matcher instead of bool) in the Rule class?
```
    fun hasLabel(key: String, value: String? = null) =
        if (value == null) {
            key in labels
        } else {
            value in getLabelValues(key)
        }
```

Reply:
@sschuberth I've extracted the IIUC uncontroversial bit to https://github.com/oss-review-toolkit/ort/pull/6084 to unblock me.
Once this is merged, I'll make a force push to this PR proposing something on top.

Reply:
> Lazy people might simply want to check for the presence of a DO_SOMETHING label instead of requiring the value to be "true" / "false" and parsing it as a boolean.

It has nothing to do with being lazy, but yes, a label does not require to have a value.

> Can we replace labelContains() and hasLabel() (see Rule.kt) with the following one...

That would make it impossible to check the exact value of a label, because it is always interpreted as a comma-separated list. Currently the user can decide how the value is interpreted so I think I would prefer to keep it as is.

Reply:
> It has nothing to do with being lazy, but yes, a label does not require to have a value.

I should have put "lazy" in quotes as it was a flippant remark to be taken with a wink.

> Currently the user can decide how the value is interpreted so I think I would prefer to keep it as is.

I tend to agree, although it's probably quite unlike that a user is using commas in the value and these commas are *not* list item delimiters.

Reply:
I've updated it and hope it now fits the preferences of both of you @mnonnenmacher @sschuberth .